// @elasticemail/elasticemail-client/dist/model/ContactsList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Elastic Email REST API
 * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
 *
 * The version of the OpenAPI document: 4.0.0
 * Contact: support@elasticemail.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ContactsList model module.
 * @module model/ContactsList
 * @version 4.0.26
 */
var ContactsList = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ContactsList</code>.
   * List of Lists, with detailed data about its contents.
   * @alias module:model/ContactsList
   */
  function ContactsList() {
    _classCallCheck(this, ContactsList);
    ContactsList.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ContactsList, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ContactsList</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ContactsList} obj Optional instance to populate.
     * @return {module:model/ContactsList} The populated <code>ContactsList</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ContactsList();
        if (data.hasOwnProperty('ListName')) {
          obj['ListName'] = _ApiClient["default"].convertToType(data['ListName'], 'String');
        }
        if (data.hasOwnProperty('PublicListID')) {
          obj['PublicListID'] = _ApiClient["default"].convertToType(data['PublicListID'], 'String');
        }
        if (data.hasOwnProperty('DateAdded')) {
          obj['DateAdded'] = _ApiClient["default"].convertToType(data['DateAdded'], 'Date');
        }
        if (data.hasOwnProperty('AllowUnsubscribe')) {
          obj['AllowUnsubscribe'] = _ApiClient["default"].convertToType(data['AllowUnsubscribe'], 'Boolean');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ContactsList</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ContactsList</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['ListName'] && !(typeof data['ListName'] === 'string' || data['ListName'] instanceof String)) {
        throw new Error("Expected the field `ListName` to be a primitive type in the JSON string but got " + data['ListName']);
      }
      // ensure the json data is a string
      if (data['PublicListID'] && !(typeof data['PublicListID'] === 'string' || data['PublicListID'] instanceof String)) {
        throw new Error("Expected the field `PublicListID` to be a primitive type in the JSON string but got " + data['PublicListID']);
      }
      return true;
    }
  }]);
}();
/**
 * Name of your list.
 * @member {String} ListName
 */
ContactsList.prototype['ListName'] = undefined;

/**
 * ID code of list. Please note that this is different from the listid field.
 * @member {String} PublicListID
 */
ContactsList.prototype['PublicListID'] = undefined;

/**
 * Date of creation in YYYY-MM-DDThh:ii:ss format
 * @member {Date} DateAdded
 */
ContactsList.prototype['DateAdded'] = undefined;

/**
 * True: Allow unsubscribing from this list. Otherwise, false
 * @member {Boolean} AllowUnsubscribe
 */
ContactsList.prototype['AllowUnsubscribe'] = undefined;
var _default = exports["default"] = ContactsList;
