// @elasticemail/elasticemail-client/dist/model/EmailStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _LogJobStatus = _interopRequireDefault(require("./LogJobStatus"));
var _MessageCategoryEnum = _interopRequireDefault(require("./MessageCategoryEnum"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Elastic Email REST API
 * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
 *
 * The version of the OpenAPI document: 4.0.0
 * Contact: support@elasticemail.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The EmailStatus model module.
 * @module model/EmailStatus
 * @version 4.0.26
 */
var EmailStatus = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EmailStatus</code>.
   * Status information of the specified email
   * @alias module:model/EmailStatus
   */
  function EmailStatus() {
    _classCallCheck(this, EmailStatus);
    EmailStatus.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(EmailStatus, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>EmailStatus</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EmailStatus} obj Optional instance to populate.
     * @return {module:model/EmailStatus} The populated <code>EmailStatus</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EmailStatus();
        if (data.hasOwnProperty('From')) {
          obj['From'] = _ApiClient["default"].convertToType(data['From'], 'String');
        }
        if (data.hasOwnProperty('To')) {
          obj['To'] = _ApiClient["default"].convertToType(data['To'], 'String');
        }
        if (data.hasOwnProperty('Date')) {
          obj['Date'] = _ApiClient["default"].convertToType(data['Date'], 'Date');
        }
        if (data.hasOwnProperty('Status')) {
          obj['Status'] = _LogJobStatus["default"].constructFromObject(data['Status']);
        }
        if (data.hasOwnProperty('StatusName')) {
          obj['StatusName'] = _ApiClient["default"].convertToType(data['StatusName'], 'String');
        }
        if (data.hasOwnProperty('StatusChangeDate')) {
          obj['StatusChangeDate'] = _ApiClient["default"].convertToType(data['StatusChangeDate'], 'Date');
        }
        if (data.hasOwnProperty('DateSent')) {
          obj['DateSent'] = _ApiClient["default"].convertToType(data['DateSent'], 'Date');
        }
        if (data.hasOwnProperty('DateOpened')) {
          obj['DateOpened'] = _ApiClient["default"].convertToType(data['DateOpened'], 'Date');
        }
        if (data.hasOwnProperty('DateClicked')) {
          obj['DateClicked'] = _ApiClient["default"].convertToType(data['DateClicked'], 'Date');
        }
        if (data.hasOwnProperty('ErrorMessage')) {
          obj['ErrorMessage'] = _ApiClient["default"].convertToType(data['ErrorMessage'], 'String');
        }
        if (data.hasOwnProperty('TransactionID')) {
          obj['TransactionID'] = _ApiClient["default"].convertToType(data['TransactionID'], 'String');
        }
        if (data.hasOwnProperty('EnvelopeFrom')) {
          obj['EnvelopeFrom'] = _ApiClient["default"].convertToType(data['EnvelopeFrom'], 'String');
        }
        if (data.hasOwnProperty('ErrorCategory')) {
          obj['ErrorCategory'] = _MessageCategoryEnum["default"].constructFromObject(data['ErrorCategory']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>EmailStatus</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>EmailStatus</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['From'] && !(typeof data['From'] === 'string' || data['From'] instanceof String)) {
        throw new Error("Expected the field `From` to be a primitive type in the JSON string but got " + data['From']);
      }
      // ensure the json data is a string
      if (data['To'] && !(typeof data['To'] === 'string' || data['To'] instanceof String)) {
        throw new Error("Expected the field `To` to be a primitive type in the JSON string but got " + data['To']);
      }
      // ensure the json data is a string
      if (data['StatusName'] && !(typeof data['StatusName'] === 'string' || data['StatusName'] instanceof String)) {
        throw new Error("Expected the field `StatusName` to be a primitive type in the JSON string but got " + data['StatusName']);
      }
      // ensure the json data is a string
      if (data['ErrorMessage'] && !(typeof data['ErrorMessage'] === 'string' || data['ErrorMessage'] instanceof String)) {
        throw new Error("Expected the field `ErrorMessage` to be a primitive type in the JSON string but got " + data['ErrorMessage']);
      }
      // ensure the json data is a string
      if (data['TransactionID'] && !(typeof data['TransactionID'] === 'string' || data['TransactionID'] instanceof String)) {
        throw new Error("Expected the field `TransactionID` to be a primitive type in the JSON string but got " + data['TransactionID']);
      }
      // ensure the json data is a string
      if (data['EnvelopeFrom'] && !(typeof data['EnvelopeFrom'] === 'string' || data['EnvelopeFrom'] instanceof String)) {
        throw new Error("Expected the field `EnvelopeFrom` to be a primitive type in the JSON string but got " + data['EnvelopeFrom']);
      }
      return true;
    }
  }]);
}();
/**
 * Email address this email was sent from.
 * @member {String} From
 */
EmailStatus.prototype['From'] = undefined;

/**
 * Email address this email was sent to.
 * @member {String} To
 */
EmailStatus.prototype['To'] = undefined;

/**
 * Date the email was submitted.
 * @member {Date} Date
 */
EmailStatus.prototype['Date'] = undefined;

/**
 * @member {module:model/LogJobStatus} Status
 */
EmailStatus.prototype['Status'] = undefined;

/**
 * Name of email's status
 * @member {String} StatusName
 */
EmailStatus.prototype['StatusName'] = undefined;

/**
 * Date of last status change.
 * @member {Date} StatusChangeDate
 */
EmailStatus.prototype['StatusChangeDate'] = undefined;

/**
 * Date when the email was sent
 * @member {Date} DateSent
 */
EmailStatus.prototype['DateSent'] = undefined;

/**
 * Date when the email changed the status to 'opened'
 * @member {Date} DateOpened
 */
EmailStatus.prototype['DateOpened'] = undefined;

/**
 * Date when the email changed the status to 'clicked'
 * @member {Date} DateClicked
 */
EmailStatus.prototype['DateClicked'] = undefined;

/**
 * Detailed error or bounced message.
 * @member {String} ErrorMessage
 */
EmailStatus.prototype['ErrorMessage'] = undefined;

/**
 * ID number of transaction
 * @member {String} TransactionID
 */
EmailStatus.prototype['TransactionID'] = undefined;

/**
 * Envelope from address
 * @member {String} EnvelopeFrom
 */
EmailStatus.prototype['EnvelopeFrom'] = undefined;

/**
 * @member {module:model/MessageCategoryEnum} ErrorCategory
 */
EmailStatus.prototype['ErrorCategory'] = undefined;
var _default = exports["default"] = EmailStatus;
