// @elasticemail/elasticemail-client/dist/model/InboundRoute.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _InboundRouteActionType = _interopRequireDefault(require("./InboundRouteActionType"));
var _InboundRouteFilterType = _interopRequireDefault(require("./InboundRouteFilterType"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Elastic Email REST API
 * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
 *
 * The version of the OpenAPI document: 4.0.0
 * Contact: support@elasticemail.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The InboundRoute model module.
 * @module model/InboundRoute
 * @version 4.0.26
 */
var InboundRoute = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InboundRoute</code>.
   * @alias module:model/InboundRoute
   */
  function InboundRoute() {
    _classCallCheck(this, InboundRoute);
    InboundRoute.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(InboundRoute, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>InboundRoute</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InboundRoute} obj Optional instance to populate.
     * @return {module:model/InboundRoute} The populated <code>InboundRoute</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InboundRoute();
        if (data.hasOwnProperty('PublicId')) {
          obj['PublicId'] = _ApiClient["default"].convertToType(data['PublicId'], 'String');
        }
        if (data.hasOwnProperty('Name')) {
          obj['Name'] = _ApiClient["default"].convertToType(data['Name'], 'String');
        }
        if (data.hasOwnProperty('FilterType')) {
          obj['FilterType'] = _InboundRouteFilterType["default"].constructFromObject(data['FilterType']);
        }
        if (data.hasOwnProperty('Filter')) {
          obj['Filter'] = _ApiClient["default"].convertToType(data['Filter'], 'String');
        }
        if (data.hasOwnProperty('ActionType')) {
          obj['ActionType'] = _InboundRouteActionType["default"].constructFromObject(data['ActionType']);
        }
        if (data.hasOwnProperty('ActionParameter')) {
          obj['ActionParameter'] = _ApiClient["default"].convertToType(data['ActionParameter'], 'String');
        }
        if (data.hasOwnProperty('SortOrder')) {
          obj['SortOrder'] = _ApiClient["default"].convertToType(data['SortOrder'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>InboundRoute</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>InboundRoute</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['PublicId'] && !(typeof data['PublicId'] === 'string' || data['PublicId'] instanceof String)) {
        throw new Error("Expected the field `PublicId` to be a primitive type in the JSON string but got " + data['PublicId']);
      }
      // ensure the json data is a string
      if (data['Name'] && !(typeof data['Name'] === 'string' || data['Name'] instanceof String)) {
        throw new Error("Expected the field `Name` to be a primitive type in the JSON string but got " + data['Name']);
      }
      // ensure the json data is a string
      if (data['Filter'] && !(typeof data['Filter'] === 'string' || data['Filter'] instanceof String)) {
        throw new Error("Expected the field `Filter` to be a primitive type in the JSON string but got " + data['Filter']);
      }
      // ensure the json data is a string
      if (data['ActionParameter'] && !(typeof data['ActionParameter'] === 'string' || data['ActionParameter'] instanceof String)) {
        throw new Error("Expected the field `ActionParameter` to be a primitive type in the JSON string but got " + data['ActionParameter']);
      }
      return true;
    }
  }]);
}();
/**
 * @member {String} PublicId
 */
InboundRoute.prototype['PublicId'] = undefined;

/**
 * Name of this route
 * @member {String} Name
 */
InboundRoute.prototype['Name'] = undefined;

/**
 * @member {module:model/InboundRouteFilterType} FilterType
 */
InboundRoute.prototype['FilterType'] = undefined;

/**
 * Filter of the inbound data
 * @member {String} Filter
 */
InboundRoute.prototype['Filter'] = undefined;

/**
 * @member {module:model/InboundRouteActionType} ActionType
 */
InboundRoute.prototype['ActionType'] = undefined;

/**
 * URL address or Email to notify about the inbound
 * @member {String} ActionParameter
 */
InboundRoute.prototype['ActionParameter'] = undefined;

/**
 * Place of this route in your routes queue's order
 * @member {Number} SortOrder
 */
InboundRoute.prototype['SortOrder'] = undefined;
var _default = exports["default"] = InboundRoute;
