// @elasticemail/elasticemail-client/dist/model/SubAccountInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AccountStatusEnum = _interopRequireDefault(require("./AccountStatusEnum"));
var _SubaccountSettingsInfo = _interopRequireDefault(require("./SubaccountSettingsInfo"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Elastic Email REST API
 * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
 *
 * The version of the OpenAPI document: 4.0.0
 * Contact: support@elasticemail.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The SubAccountInfo model module.
 * @module model/SubAccountInfo
 * @version 4.0.26
 */
var SubAccountInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SubAccountInfo</code>.
   * Detailed information about SubAccount.
   * @alias module:model/SubAccountInfo
   */
  function SubAccountInfo() {
    _classCallCheck(this, SubAccountInfo);
    SubAccountInfo.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(SubAccountInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>SubAccountInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SubAccountInfo} obj Optional instance to populate.
     * @return {module:model/SubAccountInfo} The populated <code>SubAccountInfo</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SubAccountInfo();
        if (data.hasOwnProperty('PublicAccountID')) {
          obj['PublicAccountID'] = _ApiClient["default"].convertToType(data['PublicAccountID'], 'String');
        }
        if (data.hasOwnProperty('Email')) {
          obj['Email'] = _ApiClient["default"].convertToType(data['Email'], 'String');
        }
        if (data.hasOwnProperty('Settings')) {
          obj['Settings'] = _SubaccountSettingsInfo["default"].constructFromObject(data['Settings']);
        }
        if (data.hasOwnProperty('LastActivity')) {
          obj['LastActivity'] = _ApiClient["default"].convertToType(data['LastActivity'], 'Date');
        }
        if (data.hasOwnProperty('EmailCredits')) {
          obj['EmailCredits'] = _ApiClient["default"].convertToType(data['EmailCredits'], 'Number');
        }
        if (data.hasOwnProperty('TotalEmailsSent')) {
          obj['TotalEmailsSent'] = _ApiClient["default"].convertToType(data['TotalEmailsSent'], 'Number');
        }
        if (data.hasOwnProperty('Reputation')) {
          obj['Reputation'] = _ApiClient["default"].convertToType(data['Reputation'], 'Number');
        }
        if (data.hasOwnProperty('Status')) {
          obj['Status'] = _AccountStatusEnum["default"].constructFromObject(data['Status']);
        }
        if (data.hasOwnProperty('ContactsCount')) {
          obj['ContactsCount'] = _ApiClient["default"].convertToType(data['ContactsCount'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>SubAccountInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SubAccountInfo</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['PublicAccountID'] && !(typeof data['PublicAccountID'] === 'string' || data['PublicAccountID'] instanceof String)) {
        throw new Error("Expected the field `PublicAccountID` to be a primitive type in the JSON string but got " + data['PublicAccountID']);
      }
      // ensure the json data is a string
      if (data['Email'] && !(typeof data['Email'] === 'string' || data['Email'] instanceof String)) {
        throw new Error("Expected the field `Email` to be a primitive type in the JSON string but got " + data['Email']);
      }
      // validate the optional field `Settings`
      if (data['Settings']) {
        // data not null
        _SubaccountSettingsInfo["default"].validateJSON(data['Settings']);
      }
      return true;
    }
  }]);
}();
/**
 * Public key for limited access to your Account such as contact/add so you can use it safely on public websites.
 * @member {String} PublicAccountID
 */
SubAccountInfo.prototype['PublicAccountID'] = undefined;

/**
 * Proper email address.
 * @member {String} Email
 */
SubAccountInfo.prototype['Email'] = undefined;

/**
 * @member {module:model/SubaccountSettingsInfo} Settings
 */
SubAccountInfo.prototype['Settings'] = undefined;

/**
 * Date of last activity on Account
 * @member {Date} LastActivity
 */
SubAccountInfo.prototype['LastActivity'] = undefined;

/**
 * Amount of email credits
 * @member {Number} EmailCredits
 */
SubAccountInfo.prototype['EmailCredits'] = undefined;

/**
 * Amount of emails sent from this Account
 * @member {Number} TotalEmailsSent
 */
SubAccountInfo.prototype['TotalEmailsSent'] = undefined;

/**
 * Numeric reputation
 * @member {Number} Reputation
 */
SubAccountInfo.prototype['Reputation'] = undefined;

/**
 * @member {module:model/AccountStatusEnum} Status
 */
SubAccountInfo.prototype['Status'] = undefined;

/**
 * How many contacts this SubAccount has stored
 * @member {Number} ContactsCount
 */
SubAccountInfo.prototype['ContactsCount'] = undefined;
var _default = exports["default"] = SubAccountInfo;
