// @elasticemail/elasticemail-client/dist/model/VerificationFileResultDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EmailValidationResult = _interopRequireDefault(require("./EmailValidationResult"));
var _FileUploadResult = _interopRequireDefault(require("./FileUploadResult"));
var _VerificationStatus = _interopRequireDefault(require("./VerificationStatus"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Elastic Email REST API
 * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
 *
 * The version of the OpenAPI document: 4.0.0
 * Contact: support@elasticemail.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The VerificationFileResultDetails model module.
 * @module model/VerificationFileResultDetails
 * @version 4.0.26
 */
var VerificationFileResultDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VerificationFileResultDetails</code>.
   * Detailed verification file result info
   * @alias module:model/VerificationFileResultDetails
   */
  function VerificationFileResultDetails() {
    _classCallCheck(this, VerificationFileResultDetails);
    VerificationFileResultDetails.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(VerificationFileResultDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>VerificationFileResultDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VerificationFileResultDetails} obj Optional instance to populate.
     * @return {module:model/VerificationFileResultDetails} The populated <code>VerificationFileResultDetails</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VerificationFileResultDetails();
        if (data.hasOwnProperty('VerificationResult')) {
          obj['VerificationResult'] = _ApiClient["default"].convertToType(data['VerificationResult'], [_EmailValidationResult["default"]]);
        }
        if (data.hasOwnProperty('VerificationID')) {
          obj['VerificationID'] = _ApiClient["default"].convertToType(data['VerificationID'], 'String');
        }
        if (data.hasOwnProperty('Filename')) {
          obj['Filename'] = _ApiClient["default"].convertToType(data['Filename'], 'String');
        }
        if (data.hasOwnProperty('VerificationStatus')) {
          obj['VerificationStatus'] = _VerificationStatus["default"].constructFromObject(data['VerificationStatus']);
        }
        if (data.hasOwnProperty('FileUploadResult')) {
          obj['FileUploadResult'] = _FileUploadResult["default"].constructFromObject(data['FileUploadResult']);
        }
        if (data.hasOwnProperty('DateAdded')) {
          obj['DateAdded'] = _ApiClient["default"].convertToType(data['DateAdded'], 'Date');
        }
        if (data.hasOwnProperty('Source')) {
          obj['Source'] = _ApiClient["default"].convertToType(data['Source'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>VerificationFileResultDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>VerificationFileResultDetails</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['VerificationResult']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['VerificationResult'])) {
          throw new Error("Expected the field `VerificationResult` to be an array in the JSON data but got " + data['VerificationResult']);
        }
        // validate the optional field `VerificationResult` (array)
        var _iterator = _createForOfIteratorHelper(data['VerificationResult']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _EmailValidationResult["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['VerificationID'] && !(typeof data['VerificationID'] === 'string' || data['VerificationID'] instanceof String)) {
        throw new Error("Expected the field `VerificationID` to be a primitive type in the JSON string but got " + data['VerificationID']);
      }
      // ensure the json data is a string
      if (data['Filename'] && !(typeof data['Filename'] === 'string' || data['Filename'] instanceof String)) {
        throw new Error("Expected the field `Filename` to be a primitive type in the JSON string but got " + data['Filename']);
      }
      // validate the optional field `FileUploadResult`
      if (data['FileUploadResult']) {
        // data not null
        _FileUploadResult["default"].validateJSON(data['FileUploadResult']);
      }
      // ensure the json data is a string
      if (data['Source'] && !(typeof data['Source'] === 'string' || data['Source'] instanceof String)) {
        throw new Error("Expected the field `Source` to be a primitive type in the JSON string but got " + data['Source']);
      }
      return true;
    }
  }]);
}();
/**
 * Verification result's details
 * @member {Array.<module:model/EmailValidationResult>} VerificationResult
 */
VerificationFileResultDetails.prototype['VerificationResult'] = undefined;

/**
 * Identifier of this verification result
 * @member {String} VerificationID
 */
VerificationFileResultDetails.prototype['VerificationID'] = undefined;

/**
 * Origin file name
 * @member {String} Filename
 */
VerificationFileResultDetails.prototype['Filename'] = undefined;

/**
 * @member {module:model/VerificationStatus} VerificationStatus
 */
VerificationFileResultDetails.prototype['VerificationStatus'] = undefined;

/**
 * @member {module:model/FileUploadResult} FileUploadResult
 */
VerificationFileResultDetails.prototype['FileUploadResult'] = undefined;

/**
 * Date of creation in YYYY-MM-DDThh:ii:ss format
 * @member {Date} DateAdded
 */
VerificationFileResultDetails.prototype['DateAdded'] = undefined;

/**
 * Origin file extension
 * @member {String} Source
 */
VerificationFileResultDetails.prototype['Source'] = undefined;
var _default = exports["default"] = VerificationFileResultDetails;
