// @elasticemail/elasticemail-client/dist/model/DomainData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CertificateValidationStatus = _interopRequireDefault(require("./CertificateValidationStatus"));
var _DomainOwner = _interopRequireDefault(require("./DomainOwner"));
var _TrackingType = _interopRequireDefault(require("./TrackingType"));
var _TrackingValidationStatus = _interopRequireDefault(require("./TrackingValidationStatus"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Elastic Email REST API
 * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
 *
 * The version of the OpenAPI document: 4.0.0
 * Contact: support@elasticemail.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The DomainData model module.
 * @module model/DomainData
 * @version 4.0.26
 */
var DomainData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DomainData</code>.
   * Domain data, with information about domain records.
   * @alias module:model/DomainData
   */
  function DomainData() {
    _classCallCheck(this, DomainData);
    DomainData.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(DomainData, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>DomainData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DomainData} obj Optional instance to populate.
     * @return {module:model/DomainData} The populated <code>DomainData</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DomainData();
        if (data.hasOwnProperty('ValidationLog')) {
          obj['ValidationLog'] = _ApiClient["default"].convertToType(data['ValidationLog'], 'String');
        }
        if (data.hasOwnProperty('Domain')) {
          obj['Domain'] = _ApiClient["default"].convertToType(data['Domain'], 'String');
        }
        if (data.hasOwnProperty('DefaultDomain')) {
          obj['DefaultDomain'] = _ApiClient["default"].convertToType(data['DefaultDomain'], 'Boolean');
        }
        if (data.hasOwnProperty('Spf')) {
          obj['Spf'] = _ApiClient["default"].convertToType(data['Spf'], 'Boolean');
        }
        if (data.hasOwnProperty('Dkim')) {
          obj['Dkim'] = _ApiClient["default"].convertToType(data['Dkim'], 'Boolean');
        }
        if (data.hasOwnProperty('MX')) {
          obj['MX'] = _ApiClient["default"].convertToType(data['MX'], 'Boolean');
        }
        if (data.hasOwnProperty('DMARC')) {
          obj['DMARC'] = _ApiClient["default"].convertToType(data['DMARC'], 'Boolean');
        }
        if (data.hasOwnProperty('IsRewriteDomainValid')) {
          obj['IsRewriteDomainValid'] = _ApiClient["default"].convertToType(data['IsRewriteDomainValid'], 'Boolean');
        }
        if (data.hasOwnProperty('Verify')) {
          obj['Verify'] = _ApiClient["default"].convertToType(data['Verify'], 'Boolean');
        }
        if (data.hasOwnProperty('Type')) {
          obj['Type'] = _TrackingType["default"].constructFromObject(data['Type']);
        }
        if (data.hasOwnProperty('TrackingStatus')) {
          obj['TrackingStatus'] = _TrackingValidationStatus["default"].constructFromObject(data['TrackingStatus']);
        }
        if (data.hasOwnProperty('CertificateStatus')) {
          obj['CertificateStatus'] = _CertificateValidationStatus["default"].constructFromObject(data['CertificateStatus']);
        }
        if (data.hasOwnProperty('CertificateValidationError')) {
          obj['CertificateValidationError'] = _ApiClient["default"].convertToType(data['CertificateValidationError'], 'String');
        }
        if (data.hasOwnProperty('TrackingTypeUserRequest')) {
          obj['TrackingTypeUserRequest'] = _TrackingType["default"].constructFromObject(data['TrackingTypeUserRequest']);
        }
        if (data.hasOwnProperty('VERP')) {
          obj['VERP'] = _ApiClient["default"].convertToType(data['VERP'], 'Boolean');
        }
        if (data.hasOwnProperty('CustomBouncesDomain')) {
          obj['CustomBouncesDomain'] = _ApiClient["default"].convertToType(data['CustomBouncesDomain'], 'String');
        }
        if (data.hasOwnProperty('IsCustomBouncesDomainDefault')) {
          obj['IsCustomBouncesDomainDefault'] = _ApiClient["default"].convertToType(data['IsCustomBouncesDomainDefault'], 'Boolean');
        }
        if (data.hasOwnProperty('IsMarkedForDeletion')) {
          obj['IsMarkedForDeletion'] = _ApiClient["default"].convertToType(data['IsMarkedForDeletion'], 'Boolean');
        }
        if (data.hasOwnProperty('Ownership')) {
          obj['Ownership'] = _DomainOwner["default"].constructFromObject(data['Ownership']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>DomainData</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DomainData</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['ValidationLog'] && !(typeof data['ValidationLog'] === 'string' || data['ValidationLog'] instanceof String)) {
        throw new Error("Expected the field `ValidationLog` to be a primitive type in the JSON string but got " + data['ValidationLog']);
      }
      // ensure the json data is a string
      if (data['Domain'] && !(typeof data['Domain'] === 'string' || data['Domain'] instanceof String)) {
        throw new Error("Expected the field `Domain` to be a primitive type in the JSON string but got " + data['Domain']);
      }
      // ensure the json data is a string
      if (data['CertificateValidationError'] && !(typeof data['CertificateValidationError'] === 'string' || data['CertificateValidationError'] instanceof String)) {
        throw new Error("Expected the field `CertificateValidationError` to be a primitive type in the JSON string but got " + data['CertificateValidationError']);
      }
      // ensure the json data is a string
      if (data['CustomBouncesDomain'] && !(typeof data['CustomBouncesDomain'] === 'string' || data['CustomBouncesDomain'] instanceof String)) {
        throw new Error("Expected the field `CustomBouncesDomain` to be a primitive type in the JSON string but got " + data['CustomBouncesDomain']);
      }
      return true;
    }
  }]);
}();
/**
 * Domain validation results - when domain has been running through validation process
 * @member {String} ValidationLog
 */
DomainData.prototype['ValidationLog'] = undefined;

/**
 * Name of selected domain.
 * @member {String} Domain
 */
DomainData.prototype['Domain'] = undefined;

/**
 * True, if domain is used as default. Otherwise, false,
 * @member {Boolean} DefaultDomain
 */
DomainData.prototype['DefaultDomain'] = undefined;

/**
 * True, if SPF record is verified
 * @member {Boolean} Spf
 */
DomainData.prototype['Spf'] = undefined;

/**
 * True, if DKIM record is verified
 * @member {Boolean} Dkim
 */
DomainData.prototype['Dkim'] = undefined;

/**
 * True, if MX record is verified
 * @member {Boolean} MX
 */
DomainData.prototype['MX'] = undefined;

/**
 * @member {Boolean} DMARC
 */
DomainData.prototype['DMARC'] = undefined;

/**
 * True, if tracking CNAME record is verified
 * @member {Boolean} IsRewriteDomainValid
 */
DomainData.prototype['IsRewriteDomainValid'] = undefined;

/**
 * True, if DKIM, SPF, or tracking are still to be verified
 * @member {Boolean} Verify
 */
DomainData.prototype['Verify'] = undefined;

/**
 * @member {module:model/TrackingType} Type
 */
DomainData.prototype['Type'] = undefined;

/**
 * @member {module:model/TrackingValidationStatus} TrackingStatus
 */
DomainData.prototype['TrackingStatus'] = undefined;

/**
 * @member {module:model/CertificateValidationStatus} CertificateStatus
 */
DomainData.prototype['CertificateStatus'] = undefined;

/**
 * @member {String} CertificateValidationError
 */
DomainData.prototype['CertificateValidationError'] = undefined;

/**
 * @member {module:model/TrackingType} TrackingTypeUserRequest
 */
DomainData.prototype['TrackingTypeUserRequest'] = undefined;

/**
 * @member {Boolean} VERP
 */
DomainData.prototype['VERP'] = undefined;

/**
 * @member {String} CustomBouncesDomain
 */
DomainData.prototype['CustomBouncesDomain'] = undefined;

/**
 * @member {Boolean} IsCustomBouncesDomainDefault
 */
DomainData.prototype['IsCustomBouncesDomainDefault'] = undefined;

/**
 * @member {Boolean} IsMarkedForDeletion
 */
DomainData.prototype['IsMarkedForDeletion'] = undefined;

/**
 * @member {module:model/DomainOwner} Ownership
 */
DomainData.prototype['Ownership'] = undefined;
var _default = exports["default"] = DomainData;
